### PR TITLE
tests integ: VLAN over bond interface

### PR DESCRIPTION
**Pull request by Nandan307**
Signed-off-by: Nandan Kulkarni <nakulkar@redhat.com>

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by Nandan307** [comment]
> This one looks like a duplicate of #470

Not anymore. Please resolve this comment once reviewed

File: tests/integration/bond_test.py [hunk 1]
```diff
@@ -369,3 +370,15 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
         assertlib.assert_state(bond_state)
 
     assertlib.assert_absent('bond99')
+
+
+def test_create_vlan_over_a_bond(bond99_with_slave):
+    vlan_base_iface = BOND99
```
Comment (EdDev) [inline tests/integration/bond_test.py]: Please fetch this from bond99_with_slave and not use the constant.

File: tests/integration/bond_test.py [hunk 2]
```diff
@@ -369,3 +370,15 @@ def test_bond_with_empty_ipv6_static_address(eth1_up):
...
+    vlan_id = 102
+    vlan_iface_name = '{}.{}'.format(vlan_base_iface, vlan_id)
+    with bond_interface(name='bond99', slaves=['eth2']) as bond_state:
```
Comment (EdDev) [inline tests/integration/bond_test.py]: This is what `bond99_with_slave` is doing, so you do not need it.

**Comment by Nandan307** [comment]
**Changelog:**

- Deleted ` bond_interface(name='bond99', slaves=['eth2'])` 
- assert_state now asserts `bond99_with_slave`

**Comment by EdDev** [comment]
@Nandan307 please respond to the last standing comment and make sure to rebase so we can take the change.

**Comment by EdDev** [comment]
@Nandan307 please rebase.

**Comment by Nandan307** [comment]
All set @EdDev. Thanks!

**Pull request merged**